src/Nethermind/Nethermind.Merge.Plugin/BlockProduction/BlockImprovementContext.cs

```diff
@@ -33,12 +34,12 @@ public class BlockImprovementContext : IBlockImprovementContext
         IManualBlockProductionTrigger blockProductionTrigger,
         TimeSpan timeout,
         BlockHeader parentHeader,
-        PayloadAttributes payloadAttributes,
-        DateTimeOffset startDateTime)
+        PayloadAttributes payloadAttributes)
     {
         _cancellationTokenSource = new CancellationTokenSource(timeout);
+        Watch = Stopwatch.StartNew();
```

Comment:
Maybe instead of creating Stopwatch for each improvement, we should create it only for first one, and pass stopwatch to further improvements, this way we would measure time as it was intended to and reduce object allocations a bit.